WorkflowManagement/WorkflowManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace WorkflowManagement
+{
+    public enum WorkflowStage
+    {
+        Analyze,
+        Report,
+        Telemetry,
+        Finished
+    }
+
+    /// <summary>
+    /// Workflow stages are in the following order: Analyze -> Report -> Telemetry -> Finished
+    /// </summary>
+    public class WorkflowManager
+    {
+        IWorkflowAction[] actions;
+        static WorkflowManager manager;
+
+        public static WorkflowManager GetInstance(IWorkflowActionFactory actionFactory)
```

Comment:
We us DI throughout the system so methods to manage lifetime are unnecessary

Reply:
We talked about this over IM - recapping for everyone's awareness: For now, will leave as a service locator pattern (which is an anti-pattern).  Connie will add a user story to the backlog to add proper dependency management for functions.

Reply:
I simplified my code a bit so that this singleton is no longer needed...so DI isn't needed for the time being.